.changeset/dull-lions-shave.md

```diff
@@ -0,0 +1,5 @@
+---
+'@sumup/circuit-ui': major
+---
+
+Added `UNSAFE_DISABLE_NO_MARGIN_ERRORS` environment variable to silence noMargin errors in development when it is set to `true`.
```

Comment:
What we're doing is actually throwing an error instead of a deprecation warning. The env var is just an escape hatch

```suggestion
Threw a runtime error when the `noMargin` prop isn't passed to components requiring it. Setting the `UNSAFE_DISABLE_NO_MARGIN_ERRORS` environment variable to `true` will temporarily turn off the errors.
```